### ISSUE TITLE
Add Automatic Local Backup Options with Configurable Storage and Scheduling

### ISSUE THREAD

**Issue by enrique-lozano** [enrique-lozano/Monekin#268]
## Problem

Currently, users can only create backups manually, which may lead to missed backups or data loss if the process is forgotten.

## Proposed Solution

Introduce an automatic backup feature with the following options:

- Backup Scheduling: Allow users to set backups to occur automatically at intervals such as "daily", "weekly", "monthly." or directly "on app exit".
- Custom Backup Location: Provide users with the ability to define a specific folder for storing backups instead of defaulting to the Downloads folder.

The UI in the "More" tab and in the "Settings" page may be altered by this new functionality. We will have to study carefully how to fit this new automatic backup functionality with the existing one that allows the user to save a backup at any time. 

Defining a good UI for this case seems to me perhaps one of the most complex tasks. I think we can look at [Bluecoins](https://play.google.com/store/apps/details?id=com.rammigsoftware.bluecoins) here, which offers exactly the same functionalities that we want to implement sooner or later. It has a button to download a backup on the fly (which we already have), another menu to configure automatic local backups (what we want now) and another menu to configure cloud backups (what we want in the future).

The CSV backups should remain outside of this new features since this type of exports are something that you would typically do manually from time to time.

## Next Steps

Eventually we should add support for cloud backups (e.g., Google Drive), while maintaining local backup options for users who prefer offline solutions. This way, nothing done in this task should be affected by the implementation of # (except perhaps some UI changes)

------

**Comment by victor-marino** [comment]
Hi!

I've been working a bit on this and just wanted to show a basic draft of a proposal for the UI. Let me know what you think!

Things to note:
* In the (manual) export menu, I'm thinking perhaps it's best to remove the text field with the destination folder, and just ask the user every time they press the "Export" button. It's going to be a manual action anyway, so it doesn't really need persistence, and I think it may create confusion once the automatic backups are added.
* In the automatic backups option, I thought it could be useful if we could try to do a kind of "rolling backup", where we delete older backups after creating new ones (e.g.: keep the last 3 backups, delete older ones). Otherwise we will eventually clutter the user's storage if he performs frequent backups. Not sure how hard it would be to implement this though, as we'd probably need to identify the old backups by filename. But I've added the UI for it just in case.

![data_menu](https://github.com/user-attachments/assets/cc5b036b-e562-4c6f-b1b6-48846cbdcc27)

![export_menu](https://github.com/user-attachments/assets/3b330c41-e795-4846-a840-82dddf3031ca)

![automatic_backups_off](https://github.com/user-attachments/assets/43b9c979-374e-464d-aed0-3acacfd8e80e)

![automatic_backups_on](https://github.com/user-attachments/assets/e9bbf205-8419-4f4c-ae3d-b5dbe7df58e4)

![autodelete_old_backups_on](https://github.com/user-attachments/assets/4990bfbc-8cf6-4e52-a210-03bd92a95f4b)

**Comment by enrique-lozano** [comment]
Hello! Thank you very much for the progress! I will answer you point by point:

1 - Yes, absolutely. When clicking on the export button, the user should be given the option to choose the destination directly, it makes no sense to add a previous option or persist any information. You can take a look at how [Bluecoins](https://play.google.com/store/apps/details?id=com.rammigsoftware.bluecoins) does it, the idea would be to implement the same thing. In any case, this could be part of another task since it has nothing to do with auto-backups.
2 - Wow! Very good work here! I really like the design and the option to only leave the last N backups. As a suggestion, I would say that the third screenshot shows a very empty screen, perhaps I would show the fields of the next screenshot but disabled. Apart from that, very very good work! I really like the UI! Now only the best part is missing, making it work 🙂

The "rolling backup" functionality should not be difficult, we just have to make sure that the backups are created with a name complex enough so that we do not delete other user files.

Thank you very much! Good work!

PS: For the next time, try to upload all the screenshots in English, so that the rest of the people can also read them well!

**Comment by victor-marino** [comment]
Thanks! Agreed in all your points.

Regarding screen 3, I was thinking... perhaps we can improve the layout a bit.

What if we update screen #1 ("Data") to include 3 different tile list items under the "Export" section:
* Export data (already there)
* Local backups
* Cloud backups (in the future)

In the last two items, we could already show their status with something like "Active" (in green) "Not active" (black) in the subtitle:

![backup_status](https://github.com/user-attachments/assets/b0d63e89-98b6-4f6f-8159-e4b87f4d7ca2)

This way, once the user taps in each of the options, we can show all the corresponding settings for that kind of backup without the need to hide or disable anything, as the screen will be dedicated to that kind of backup only.

What do you think?

**Comment by enrique-lozano** [comment]
Hi @victor-marino! Yes, I like the idea, looks good for me 🚀

**Comment by victor-marino** [comment]
Hey there!

Sorry for the absence of updates, was really busy lately. I think I should have some time to work on Monekin over the next few weeks.

If you agree, I think the first logical step (and the easiest to implement) is to simply allow the user to choose the folder where they want to save the existing (manual) backups. Once that's in place we can start working on the scheduling.

I will submit a PR for this shortly!

**Comment by enrique-lozano** [comment]
Agree! We can merge the PR with this functionality as this is completely independent from the other part

**Comment by victor-marino** [comment]
Great, just submitted the PR: https://github.com/enrique-lozano/Monekin/pull/302
